### ISSUE TITLE
Add mapping layer to GitHub

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#575]
Make it so people can log in with GitHub without having to use their GitHub account name.

Turns out, it's fairly common to allow people to use OAuth with a mapping layer so they don't have to have the same name on two different services. Doing this will allow me to continue to be `deadhead` on mb2 (I'm `ctm` on GitHub).

I'm tagging this as `AGDGAT` but am not including it on this week's milestone because I'm behind due to my vBMDM work.

**Comment by ctm** [comment]
FWIW, due to the lack of this feature, I stopped using the GitHub functionality a long time ago and inadvertently broke it (#788) with an `actix-web` upgrade an then broke it further when I updated to `yew` 0.19.

So, at a minimum I should change the database schema to accommodate this mapping, even if I initially need to hand-populate the mapping that will allow me to use `deadhead` via `ctm`'s credentials.

**Comment by ctm** [comment]
Looks like the "cleanest" (/me coughs) solution would be for the GitHub handling code in main to not look for the match between the username handed back from GitHub and the desired nickname but to instead pass both of those into `NickMapper`. Main does not have access to the database (nor should it) and `NickMapper`'s original job was to provide mappings for nicknames.

So, just add a nullable string `github_user` field to the `players` table and then use that to identify the mb2 player. So, instead of:
```
                            if let Err(e) = recipient.do_send(GitHubOAuthResponse(nick, firefox)) {
...
```
we'll have something like:
```
                            if let Err(e) = recipient.do_send(GitHubOAuthResponse(desired_nick, details.login, firefox)) {
...
```
and all the code from there until we do the db lookup will have to not care that `desired_nick` might be `None`.

Whee!

**Comment by ctm** [comment]
I made a lot of progress on this today. If the db is properly configured, the right thing happens.

However, the code that creates GitHub accounts is broken in my new branch and is probably broken in production. Fixing it shouldn't be particularly hard, so if I'm not distracted tomorrow I should get it done then.

**Comment by ctm** [comment]
Done. Deploying now.

**Comment by ctm** [comment]
Ugh. I accidentally deleted some of this functionality when I was working on the wrong branch earlier today. I'm pretty sure I can get it back via `git reflog`.

Whee! **Not!**

**Comment by ctm** [comment]
Fixed. Deploying now.

In case anyone (including future `deadhead`) is wondering, I had `master` checked out earlier this morning due to upgrading crates. I also had a little list of things that I needed to do in order to complete the missing functionality. I went about my work until I finally got to a point where I realized I was working on the wrong branch. I did a `git stash`, `git checkout gh-map-575` and a `git stash pop`. The pop failed with conflicts and I didn't catch a change.

Later Rust was warning me that I had a function `nick_for_github_user` that was not used and even then I still didn't snap. Instead I poked around a bit and saw that it really wasn't used, so I deleted it

When I was done with my mods, I tested them, but didn't test the mods I had made much earlier, because almost all of those mods were being tested each time I tested the new mods. However, almost isn't all. Once I deployed to `devctm.com` and tried to log in as `deadhead` via the GitHub button, it tried to log me in as `ctm`.

Anyway, `git reflog` did indeed have the missing code and it was trivial to invoke it. Deployment takes a long time, so I decided to write this final (I hope!) note before eating my dinner.
